engine/src/test/java/io/camunda/zeebe/engine/processing/resource/ResourceDeletionTest.java

```diff
@@ -261,6 +264,69 @@ public void shouldCreateInstanceOfVersionTwoWhenVersionOneIsDeleted() {
     verifyInstanceOfProcessWithIdAndVersionIsCompleted(processId, 2, processInstanceKey);
   }
 
+  @Test
+  public void shouldCancelTimerStartEventOnDeletion() {
+    // given
+    final var processId = helper.getBpmnProcessId();
+    final var processDefinitionKey = deployProcessWithTimerStartEvent(processId);
+
+    // when
+    engine.resourceDeletion().withResourceKey(processDefinitionKey).delete();
+
+    // then
+    verifyTimerIsCancelled(processDefinitionKey);
+    verifyProcessIdWithVersionIsDeleted(processId, 1);
+    verifyResourceIsDeleted(processDefinitionKey);
+  }
+
+  @Test
+  public void shouldCancelAllTimerStartEventsOnDeletion() {
+    // given
+    final var processId = helper.getBpmnProcessId();
+    final var processDefinitionKey =
+        engine
+            .deployment()
+            .withXmlResource(
+                Bpmn.createExecutableProcess(processId)
+                    .startEvent("startEvent1")
+                    .timerWithDuration(Duration.ofDays(1))
+                    .endEvent("endEvent")
+                    .moveToProcess(processId)
+                    .startEvent("startEvent2")
+                    .timerWithDuration(Duration.ofDays(1))
+                    .connectTo("endEvent")
+                    .done())
+            .deploy()
+            .getValue()
+            .getProcessesMetadata()
+            .get(0)
+            .getProcessDefinitionKey();
+
+    // when
+    engine.resourceDeletion().withResourceKey(processDefinitionKey).delete();
+
+    // then
+    verifyTimersAreCancelled(processDefinitionKey, 2);
+    verifyProcessIdWithVersionIsDeleted(processId, 1);
+    verifyResourceIsDeleted(processDefinitionKey);
+  }
+
+  @Test
+  public void shouldNotCancelTimersIfDeletedVersionIsNotLatest() {
```

Comment:
💭 This is not completely true right ? I give you an example. 

_We deploy two version of a process (both with a timer start event), the v1 timer is deleted due to the deploy of the v2 process._

This test succeed only because the first process doesn't contain a timer start event. I don't understand the purpose of this test.
All of this just to say that the name of this test is a bit confusing (at least for me 😂)

Reply:
The name says should**Not**CancelTimersIfDeletedVersionIs**Not**Latest

The scenario is:
- We deploy process V1 without timer start event
- We deploy process V2 with timer start event
- We delete V1
- We verify that the timers of V2 are not getting deleted

The goal of this test is verifying we don't touch the timers of V2 when we delete an older version of this process. Does this clarify it?

Reply:
Yes, got it! At first glance it may suggest that timers are never deleted if they are not from the latest version (referred also to my slack message). But that's because I didn't know how deploying new versions works.